code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm

```diff
@@ -11,626 +11,549 @@
-				if("fuel")
-					icon_state = "dr_gibb_glass"
-					name = "Glass of welder fuel"
-					desc = "Unless you are an industrial tool, this is probably not safe for consumption."
-				if("brownstar")
-					icon_state = "brownstar"
-					name = "Brown Star"
-					desc = "Its not what it sounds like..."
-				if("tea")
-					icon_state = "glass_brown"
-					name = "Glass of Tea"
-					desc = "A glass of hot tea. Perhaps a cup with a handle would have been smarter?"
-				if("icetea")
-					icon_state = "icetea"
-					name = "Iced Tea"
-					desc = "No relation to a certain rap artist/ actor."
-				if("milkshake")
-					icon_state = "milkshake"
-					name = "Milkshake"
-					desc = "Glorious brainfreezing mixture."
-				if("lemonade")
-					icon_state = "lemonade"
-					name = "Lemonade"
-					desc = "Oh the nostalgia..."
-				if("kiraspecial")
-					icon_state = "kiraspecial"
-					name = "Kira Special"
-					desc = "Long live the guy who everyone had mistaken for a girl. Baka!"
-				if("rewriter")
-					icon_state = "rewriter"
-					name = "Rewriter"
-					desc = "The secert of the sanctuary of the Libarian..."
-				if("applejack")
-					icon_state = "cognacglass"
-					name = "Glass of applejack"
-					desc = "When cider isn't strong enough, you gotta jack it."
-				if("jackrose")
-					icon_state = "patronglass"
-					name = "Jack Rose"
-					desc = "Drinking this makes you feel like you belong in a luxury hotel bar during the 1920s."
-				if("synthanol")
-					icon_state = "synthanolglass"
-					name = "Glass of Synthanol"
-					desc = "The equivalent of alcohol for synthetic crewmembers. They'd find it awful if they had tastebuds too."
-				if("robottears")
-					icon_state = "robottearsglass"
-					name = "Glass of Robot Tears"
-					desc = "No robots were hurt in the making of this drink."
-				if("trinary")
-					icon_state = "trinaryglass"
-					name = "Glass of Trinary"
-					desc = "Colorful drink made for synthetic crewmembers. It doesn't seem like it would taste well."
-				if("servo")
-					icon_state = "servoglass"
-					name = "Glass of Servo"
-					desc = "Chocolate - based drink made for IPCs. Not sure if anyone's actually tried out the recipe."
-				if("synthnsoda")
-					icon_state = "synthnsodaglass"
-					name = "Glass of Synth 'n Soda"
-					desc = "Classic drink altered to fit the tastes of a robot. Bad idea to drink if you're made of carbon."
-				if("synthignon")
-					icon_state = "synthignonglass"
-					name = "Glass of Synthignon"
-					desc = "Someone mixed good wine and robot booze. Romantic, but atrocious."
-				if("uplink")
-					icon_state = "uplinkglass"
-					name = "Glass of Uplink"
-					desc = "An exquisite mix of the finest liquoirs and synthanol. Meant only for synthetics."
-				if("holywater")
-					icon_state = "glass_clear"
-					name = "Glass of Water"
-					desc = "The father of all refreshments."
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/on_reagent_change() // *scream

```

Comment:
ideally, it would be based on the reagent datum, itself and the glass would just set its icon state to the master reagent's "glass_icon" or something like that. Definitely not doing that in this PR.